picodom.d.ts

```diff
@@ -0,0 +1,30 @@
+declare namespace picodom {
+  export interface VNode<D> {
+    tag: string
+    data?: D
+    children: VNode<{}>
+  }
+
+  export interface Component<D> {
+    (data?: D, ...children: VNode<{}>[]): VNode<D>
```

Comment:
Similar to [Optional Parameters in Callbacks](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks), I don't think data needs to be optional here. The Component interface will always be invoked with a `data` argument and `children` argument. If a consumer wants to ignore them that is valid.

I tested the following locally and it passes:
```typescript
export interface Component<D> {
  (data: D, children: VNode<{}>[]): VNode<D>
}

let Test: Component<any> = () => <div></div>;
```